client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx

```diff
@@ -0,0 +1,98 @@
+import { Button, Card } from '@automattic/components';
+import { getQueryArg } from '@wordpress/url';
+import { useTranslate } from 'i18n-calypso';
+import page from 'page';
+import { ReactElement, useCallback, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import FormRadio from 'calypso/components/forms/form-radio';
+import SearchCard from 'calypso/components/search-card';
+import { addQueryArgs } from 'calypso/lib/url';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { errorNotice, infoNotice } from 'calypso/state/notices/actions';
+import useAssignLicenseMutation from 'calypso/state/partner-portal/licenses/hooks/use-assign-license-mutation';
+import './style.scss';
+
+export default function AssignLicenseForm( { sites }: any ): ReactElement {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const [ filter, setFilter ] = useState( false );
+	const [ selectedSite, setSelectedSite ] = useState( false );
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const licenseKey = getQueryArg( window.location.href, 'key' ) as string;
+	const onSelectSite = ( site: any ) => setSelectedSite( site );
+
+	const siteCards = sites.map( ( site: any ) => {
+		if ( -1 !== site.domain.search( filter ) || false === filter ) {
+			return (
+				<Card key={ site.ID } className="assign-license-form__site-card">
+					<FormRadio
+						className="assign-license-form__site-card-radio"
+						label=""
+						name="site_select"
+						onClick={ () => onSelectSite( site.ID ) }
+					/>
+					{ site.domain }
+				</Card>
+			);
+		}
+	} );
+
+	const onSearch = ( query: any ) => setFilter( query );
```

Comment:
We can skip this and pass setFilter directly as the event callback. That will also avoid some unnecessary renders as the value won't change.

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/60903/commits/9035c9ea6329895054bafb6d41f11392d381d966.